lm_eval/tasks/pie_perf.py

```diff
@@ -0,0 +1,133 @@
+        return self.dataset["codegen_1shot_test"]
+
+    def fewshot_examples(self):
+        """Loads and returns the few-shot examples for the task if they exist."""
+        pass
+
+    def get_prompt(self, doc):
+        """
+        Builds the prompt for the LM to generate from.
+        :param doc: dict[str: str]
+            sample from the test dataset
+        :return: str
+        """
+        prompt = doc['input']
+        return prompt
+
+    def get_reference(self, doc):
+        """
+        Builds the reference solution for the doc (sample from the test dataset).
+        :param doc: dict[str: str]
+            sample from the test dataset
+        :return: str
+        """
+        return doc['target']
+
+    def postprocess_generation(self, generation, idx):
+        """
+        Defines the postprocessing for a LM generation.
+        :param generation: str
+            code generation from LM
+        :param idx: int (if needed)
+            index of doc in the dataset to which the generation belongs
+        :return: str
+        """
+        question_sep = "# slower version:"
+        answer_sep = "# optimized version of the same code:"
+
+        output_code = generation.replace("\n\n\n\n\n", "")
+        output_code = output_code.split(question_sep)[-1]
+        output_code = output_code.split(answer_sep)[-1].strip()
+
+        problem_id = self.dataset['codegen_1shot_test']['problem_id'][idx]
+        submission_id = self.dataset['codegen_1shot_test']['submission_id_v0'][idx]
+
+        return output_code, problem_id, submission_id
+
+    def process_results(self, generations, references):
+        """
+        Takes the list of LM generations and evaluates them against ground truth references,
+        returning the metric for the generations as in {"metric_name": result}.
+        We encourage to directly load the metric from `evaluate` library to keep the code concise.
+        :param generations: list(list(str))
+            list of lists containing generations
+        :param references: list(str)
+            list of str containing refrences
+        :return: dict[str: float]
+        """
+        limit = len(references)
+
+        # preparing testcases
+        cmd = "git clone https://huggingface.co/datasets/rootacess/pie-perf-testcases lm_eval/tasks/custom_metrics/pie_perf_metric/public_test_cases"
+        process = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        logging.error(f'An error occurred: {error}')
+
+        # running evaluations
+        res = compute(generations, references, dataset=self.get_dataset()[:limit])
+
+        # cleaning up
+        cmd = "rm -rf lm_eval/tasks/custom_metrics/pie_perf_metric/public_test_cases"
+        process = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        logging.error(error)
```

Comment:
Can't we load the test cases with HF datasets?
If not at least should check that the path doesn't already exist I think

Reply:
Due to the invariable numbers of test cases per type of problem, it is not suitable to convert it into Dataset format. I can add the condition to check if path exists